### ISSUE TITLE
Setting target runner when running from an archive

### ISSUE THREAD

**Issue by pascalkuthe** [nextest-rs/nextest#468]
I am using cargo nextest to run cross compiled builds for `x86_64-pc-windows-msvc` using wine.
To achieve I have set `CARGO_TARGET_X86_64_PC_WINDOWS_MSVC_RUNNER="wine"`.

Running without reusing builds works as expected: 

```
cargo nextest run --target x86_64-pc-windows-msvc
```

However, I want to run tests across multiple containers in parallel and reuse builds across them.
My workflow looks roughly like this:

```
cargo nextest archive --target x86_64-pc-windows-msvc --archive-file=foo.tar.zst
cargo nextest run --archive-file=foo.tar.zst
```

However this does currently not work because the second invocation does not respect `CARGO_TARGET_X86_64_PC_WINDOWS_MSVC_RUNNER="wine"`.

I have tried digging into what is going on exactly but I am not quite sure.
I found that setting `CARGO_BUILD_TARGET=x86_64-pc-windows-msvc` is a workaround.
That would suggest the problem is simply that cargo nextest is using the wrong target for running the tests.
However specifying `--target` is not allowed when `--archive-file` is given.
Furthermore setting `CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUNNER="wine"` (or the same for MUSL) does not seem to be respected either. From glancing at the sourcecode I am guessing without this manual flag the target-triple is set to `None` (because no build actually runs?) and therefore no runner is actually used.

To me the a solution seems to be one of:
* Allow passing `--target` to `cargo nextest run` even if `--archive` is given
* Save the target into the archive and respect it properly during `cargo nextest run`
* (Using some existing way to detect the correct target?)

If there is consensus on how to resolve this, I could send a PR if that would be desired although I am not super familiar with the codebase yet

**Comment by sunshowers** [comment]
Thanks! I think saving the target into the archive (specifically in the rust build metadata: https://docs.rs/nextest-metadata/latest/nextest_metadata/struct.RustBuildMetaSummary.html) is the correct approach here.

You're welcome to send a PR, or I can look at this in the next few days.

**Comment by sunshowers** [comment]
(this is the first time using both archives and a target runner has come up, so thank you for digging into this!)

**Comment by pascalkuthe** [comment]
Thanks for pointing me in the right direction :) I will try to send a PR today

**Comment by sunshowers** [comment]
@pascalkuthe I added an integration test. Going to get a new release out with this tomorrow, hopefully.

**Comment by sunshowers** [comment]
Just released cargo-nextest 0.9.35 with these changes. Binaries should be available within 20-30 minutes.

**Comment by pascalkuthe** [comment]
@sunshowers  Great that you managed to get a testsetup and release going so quickly! cargo-nextest has been super helpful for me, thanks for maintaining it!
